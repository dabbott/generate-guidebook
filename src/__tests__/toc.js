const toc = require('../toc')

const sample = `
# H1 a

Other text

# H1 b

## H2 b 1

### H3 b 1 i

Some text

### H3 b 1 ii

## H2 c 1

Some text
`

test('parses headings', () => {
  const result = toc(sample)

  expect(result).toMatchSnapshot()
})

const mdSample = `
# Foo **strong**

# Bar \`code\`

# Baz *emphasis*
`

test('parses headings with formatting markdown', () => {
  const result = toc(mdSample)

  expect(result).toMatchSnapshot()
})

const mdxSample = `
# Foo <b>Test</b>
`

test('parses headings with JSX elements', () => {
  const result = toc(mdxSample)

  expect(result).toMatchSnapshot()
})
