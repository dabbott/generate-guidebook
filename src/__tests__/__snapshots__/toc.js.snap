// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parses headings 1`] = `
Array [
  Object {
    "level": 1,
    "title": "H1 a",
    "url": "#h1-a",
  },
  Object {
    "level": 1,
    "title": "H1 b",
    "url": "#h1-b",
  },
  Object {
    "level": 2,
    "title": "H2 b 1",
    "url": "#h2-b-1",
  },
  Object {
    "level": 3,
    "title": "H3 b 1 i",
    "url": "#h3-b-1-i",
  },
  Object {
    "level": 3,
    "title": "H3 b 1 ii",
    "url": "#h3-b-1-ii",
  },
  Object {
    "level": 2,
    "title": "H2 c 1",
    "url": "#h2-c-1",
  },
]
`;

exports[`parses headings with JSX elements 1`] = `
Array [
  Object {
    "level": 1,
    "title": "Foo Test",
    "url": "#foo-btestb",
  },
]
`;

exports[`parses headings with formatting markdown 1`] = `
Array [
  Object {
    "level": 1,
    "title": "Foo strong",
    "url": "#foo-strong",
  },
  Object {
    "level": 1,
    "title": "Bar code",
    "url": "#bar-code",
  },
  Object {
    "level": 1,
    "title": "Baz emphasis",
    "url": "#baz-emphasis",
  },
]
`;
